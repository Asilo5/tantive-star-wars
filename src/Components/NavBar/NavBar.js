 import React from 'react';
 import './NavBar.scss';
 import PropTypes from 'prop-types';
 import { NavLink } from 'react-router-dom';

 const NavBar = ({userName, userQuote, userRank, favCount, resetOnLogOut }) => {
     return (
         <nav>
            <div>
                <img src='https://am22.akamaized.net/tms/cnt/uploads/2017/08/leiatop1-650x574.jpg' alt='Princess Leia' />
                <h2>{userName}</h2>
                <h3>"{userQuote}"</h3>
                <h3>{userRank}</h3>
            </div>
            <div className='nav-links'>
                <NavLink to='/movies' className='nav' activeClassName='active-nav' >Movies</NavLink>
                <NavLink to='/favourites' className='nav' activeClassName='active-nav' >Favourites {favCount}</NavLink>
                <NavLink exact to='/' > <button className='nav' onClick={() => resetOnLogOut()}> Log Out </button> </NavLink>
            </div>
         </nav> 
     )
 }

NavBar.propTypes = {
    userName: PropTypes.string, 
    userQuote: PropTypes.string, 
    userRank: PropTypes.string, 
    favCount: PropTypes.number, 
    resetOnLogOut: PropTypes.func
}

export default NavBar;