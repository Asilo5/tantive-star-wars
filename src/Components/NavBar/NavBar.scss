$primary-color : #00D800;

nav {
    background: transparent;
    border-right: 1px solid $primary-color;
    height: auto;
    display: flex;
    flex-direction: column;
    padding: 10px;
    max-width: 18%;

    img {
        height: 11em;
        padding: 5px;
        border-radius: 25px;
        margin: 5px;
        border: 1px solid $primary-color;
    }

    h2 {
      color: $primary-color;
    }

    h3 {
        color: $primary-color;
    }
}

.nav-links {
    display: flex;
    flex-direction: column;
    padding: 30px;

    .nav {
        color: $primary-color;
        margin: 30px;
        font-size: 25px;
        text-decoration: none;
        cursor: pointer;
        padding: 5px;
        &:hover {
           border-top: 1px solid $primary-color;
           border-bottom: 1px solid $primary-color;
        }
    }

    .active-nav {
        border-top: 1px solid $primary-color;
        border-bottom: 1px solid $primary-color;
    }
}

#blockColorblindContent {
    display: none;
}