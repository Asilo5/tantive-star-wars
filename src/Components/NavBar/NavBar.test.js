import React from 'React';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

describe('NavBar', () => {

    let wrapper = shallow(<NavBar userName = 'Consuelo'
                                  userQuote = 'Luke, Ich bin dein Vater'
                                  userRank = 'Novice' />);

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

})