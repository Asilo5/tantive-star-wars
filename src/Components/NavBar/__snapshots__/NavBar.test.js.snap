// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot with all information passing in correctly 1`] = `
<nav>
  <div>
    <img
      alt="Princess Leia"
      src="https://am22.akamaized.net/tms/cnt/uploads/2017/08/leiatop1-650x574.jpg"
    />
    <h2>
      Consuelo
    </h2>
    <h3>
      "
      Luke, Ich bin dein Vater
      "
    </h3>
    <h3>
      Novice
    </h3>
  </div>
  <div
    className="nav-links"
  >
    <NavLink
      activeClassName="active-nav"
      className="nav"
      to="/movies"
    >
      Movies
    </NavLink>
    <NavLink
      activeClassName="active-nav"
      className="nav"
      to="/favourites"
    >
      Favourites 
    </NavLink>
    <NavLink
      exact={true}
      to="/"
    >
       
      <button
        className="nav"
        onClick={[Function]}
      >
         Log Out 
      </button>
       
    </NavLink>
  </div>
</nav>
`;
